src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -38,55 +38,46 @@ public SecretMasker() : this(default, default, default, default, default)
     {
     }
 
-    public SecretMasker(IEnumerable<RegexPattern>? regexSecrets,
+    public SecretMasker(IEnumerable<RegexPattern>? regexSecrets = null,
                         bool generateCorrelatingIds = false,
                         IRegexEngine? regexEngine = default,
                         string? defaultRegexRedactionToken = null,
                         string? defaultLiteralRedactionToken = null)
     {
-        _disposed = false;
-
-        RegexPatterns = regexSecrets != null
-            ? new HashSet<RegexPattern>(regexSecrets)
-            : new HashSet<RegexPattern>();
-
+        _regexPatterns = new HashSet<RegexPattern>(regexSecrets ?? []);
         _generateCorrelatingIds = generateCorrelatingIds;
 
-        ExplicitlyAddedSecretLiterals = new HashSet<SecretLiteral>();
-        EncodedSecretLiterals = new HashSet<SecretLiteral>();
-        LiteralEncoders = new HashSet<LiteralEncoder>();
+        _explicitlyAddedSecretLiterals = new HashSet<SecretLiteral>();
+        _encodedSecretLiterals = new HashSet<SecretLiteral>();
+        _literalEncoders = new HashSet<LiteralEncoder>();
 
         _regexEngine = regexEngine ??= CachedDotNetRegex.Instance;
 
         DefaultRegexRedactionToken = defaultRegexRedactionToken ?? RegexPattern.FallbackRedactionToken;
         DefaultLiteralRedactionToken = defaultLiteralRedactionToken ?? SecretLiteral.FallbackRedactionToken;
 
-        AddHighPerformancePatterns(RegexPatterns);
+        AddHighPerformancePatterns(_regexPatterns);
     }
 
-    // We don't permit secrets great than 5 characters in length to be
-    // skipped at masking time. The secrets that will be ignored when
-    // masking will N - 1 of this property value.
-    public static int MinimumSecretLengthCeiling { get; set; }
-
     [ThreadStatic]
     private static StringBuilder? s_stringBuilder;
 
-    public virtual string DefaultRegexRedactionToken { get; set; }
+    public string DefaultRegexRedactionToken { get; }
 
-    public virtual string DefaultLiteralRedactionToken { get; set; }
+    public string DefaultLiteralRedactionToken { get; }
 
-    public virtual HashSet<RegexPattern> RegexPatterns { get; protected set; }
+    private readonly HashSet<RegexPattern> _regexPatterns;
 
     /// <summary>
-    /// Gets the total time in ticks spent masking content for the lifetime of this masker instance.
+    /// Gets the total time spent masking content for the lifetime of this masker instance.
     /// </summary>
-    public long ElapsedMaskingTime { get; private set; }
+    public TimeSpan ElapsedMaskingTime => TimeSpan.FromTicks(_elapsedMaskingTicks);
+    private long _elapsedMaskingTicks;
 
     /// <summary>
     /// This implementation assumes no more than one thread is adding regexes, values, or encoders at any given time.
```

Comment:
>This implementation assumes no more than one thread is adding regexe [](http://example.com/codeflow?start=8&length=68)

i don't think this comment is correct any longer due to changes you have made, yes?

Reply:
Correct. I thought I got all of these. Will fix.